test/e2e/common/pods.go

```diff
@@ -772,4 +772,83 @@ var _ = framework.KubeDescribe("Pods", func() {
 		validatePodReadiness(false)
 
 	})
+
+	It("pod should not be available after the graceful terimination delete call", func() {
+		pod := getProcessedPod(podClient, f)
+		By("waiting for pod get removed from the pods list")
+		Expect(wait.Poll(time.Second*5, time.Second*10, func() (bool, error) {
```

Comment:
The pod is terminated gracefully in 30 seconds at line 851.
Why here is polling the pod existence in 10 seconds only?

Reply:
To prove pod will be removed from the pod list before the pod get terminated gracefully

Reply:
humm, I cannot understand the above meaning.
line 847 terminates a pod gracefully within 30 seconds, IIUC that means "it is fine that the pod finished in 30 seconds, after that the pod will be enforced to be terminated".
So that seems fine even if the pod still lives after 10 seconds.

Reply:
If you delete a pod gracefully, it should wait for the process completes until the graceful period seconds. But pod resource will be removed before the graceful termination period, so you can not control the pod after the graceful termination call.

Reply:
@oomichi

Reply:
Is that behavior based on the implementation?
The API reference (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#pod-v1-core) doesn't mention the behavior.
I am digging the implementation, happy to see some reference if possible.